website/pages/guides/hcl/variables.mdx

```diff
@@ -18,54 +18,124 @@ Local variables can be a compound of input variables and local variables.
 
 ## Defining Variables and locals
 
-Let's create a file `variables.pkr.hcl` with the following contents.
+In the legacy JSON packer templates, any variables we hadn't already defined in
+the "variables" stanza of our json template could simply be passed in via the
+command line or a var-file; if a variable was never defined it would generally
+be interpolated to an empty string.
+
+*In the HCL2 packer templates, we must always pre-define our variables in the
+HCL equivalent of the "variables" stanza.*
+
+Another difference between JSON and HCL packer templates is that in JSON packer
+templates, the "variables" stanza, if used, was always in the same .json file
+as the builds and builders. In HCL, it can exist in its own file if you want it
+to.
+
+To demonstrate, let's create a file `variables.pkr.hcl` with the following
+contents.
 
 -> **Note**: that the file can be named anything, since Packer loads all
 files ending in `.pkr.hcl` in a directory. If you split your configuration
-across multiple files, use `packer build <source directory>` to initiate
-a build.
+across multiple files, use
+`packer build <command line flags> <source directory>` to initiate a build.
 
 ```hcl
-# variables.pkr.hcl
+//  variables.pkr.hcl
+
+// For those variables that you don't provide a default for, you must
+// set them from the command line, a var-file, or the environment.
```

Comment:
I recall the var-file only supporting JSON files at one point so I checked to see what our help text said about the var-file flag and found that they also need to be updated 

```
packer validate --help
....
  -syntax-only           Only check syntax. Do not verify config of the template.                                                                 
  -except=foo,bar,baz    Validate all builds other than these.
  -only=foo,bar,baz      Validate only these builds.
  -var 'key=value'       Variable for templates, can be used multiple times.                                                                      
  -var-file=path         JSON file containing user variables. [ Note that even in HCL mode this expects file to contain JSON, a fix is comming soon
```

```
packer build --help
...
  -timestamp-ui                 Enable prefixing of each ui output with an RFC3339 timestamp.
  -var 'key=value'              Variable for templates, can be used multiple times.
  -var-file=path                JSON file containing user variables.
```